process/smartContract/builtInFunctions/esdtBurn.go

```diff
@@ -45,6 +45,11 @@ func NewESDTBurnFunc(
 	return e, nil
 }
 
+// SetNewGasConfig is called whenever gas cost is changed
+func (e *esdtBurn) SetNewGasConfig(gasCost *process.GasCost) {
+	e.funcGasCost = gasCost.BuiltInCost.ESDTBurn
```

Comment:
We need to protect `e.funcGasCost` with a mutex

Reply:
no need. there is no possibility to call the from different go routines

Reply:
it is protected all the way right now.